### ISSUE TITLE
Final output of U-net is SiLU

### ISSUE THREAD

**Issue by Bulbatronik** [Julian-Wyatt/AnoDDPM#39]
Hi, I was checking that your U-Net has the following structure:

```python
# UNet.py, line 384
self.out = nn.Sequential(
                GroupNorm32(32, ch),
                nn.SiLU(),
                zero_module(nn.Conv2d(base_channels * channel_mults[0], self.out_channels, 3, padding=1))
                )
```
However, for instance, CIFAR-10 is preprocessed  to keep the pixel values within [-1, 1] in the following way:
```python
# dataset.py, line 793
def load_CIFAR10(args, train=True):
    return torch.utils.data.DataLoader(
            datasets.CIFAR10(
                    "./DATASETS/CIFAR10", train=train, download=True, transform=transforms
                        .Compose(
                            [
                                transforms.ToTensor(),
                                transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))

                                ]
                            )
                    ),
            shuffle=True, batch_size=args["Batch_Size"], drop_last=True
            )
```
Shouldn't the final activation be **tangent** in this case, as to make sure the output of the UNet are in the same way? Or perhaps the values need to be rescaled within [0, 1]?

**Comment by Julian-Wyatt** [comment]
Hi, 

Thanks for checking the work!
That specific case comes from the GuidedDiffusion repository from OpenAI.

However, for an intuition about why this works, the diffusion model predicts the noise, eps, added to the image at training time and this can be unbounded (Try checking torch.randn(1,1,256,256).max()). Tanh would restrict this to [-1,1] causing unstable weighting in gradients and the final convolution.

The model head still outputs with learned mu=0 sig=0.5 due to the learned bias from the conv.
